manager/dispatcher/dispatcher.go

```diff
@@ -572,20 +573,45 @@ func (d *Dispatcher) Tasks(r *api.TasksRequest, stream api.Dispatcher_TasksServe
 			return err
 		}
 
-		select {
-		case event := <-nodeTasks:
-			switch v := event.(type) {
-			case state.EventCreateTask:
-				tasksMap[v.Task.ID] = v.Task
-			case state.EventUpdateTask:
-				tasksMap[v.Task.ID] = v.Task
-			case state.EventDeleteTask:
-				delete(tasksMap, v.Task.ID)
+		// bursty events should be processed in batches and sent out snapshot
+		const eventBatchLimit = 200
+		const eventPausedGap = 50 * time.Millisecond
+		var eventCnt int
+		// eventPaused is true when there are events
+		// but next event has not arrived within eventPausedGap
+		eventPaused := false
+
+		for eventCnt < eventBatchLimit && !eventPaused {
+			select {
+			case event := <-nodeTasks:
```

Comment:
@aluzzardi: It depends what behavior you want. Commit events will tie the task dispatching batches to the scheduler batches. That may or may not be what we want. I think either time-based or commit-based batching will be an improvement over what we have now, but they won't act quite the same way.